modules/optimization/src/base/OptimizationApp.C

```diff
@@ -36,21 +36,16 @@ OptimizationApp::OptimizationApp(InputParameters parameters) : MooseApp(paramete
 OptimizationApp::~OptimizationApp() {}
 
 void
-OptimizationApp::registerAll(Factory & f, ActionFactory & af, Syntax & s)
+OptimizationApp::registerAll(Factory & f, ActionFactory & af, Syntax & syntax)
 {
   Registry::registerObjectsTo(f, {"OptimizationApp"});
   Registry::registerActionsTo(af, {"OptimizationApp"});
 
-  auto & syntax = s;
-
   // Optimization reporter actions
   registerSyntaxTask(
       "AddOptimizationReporterAction", "OptimizationReporter", "add_optimization_reporter");
   registerMooseObjectTask("add_optimization_reporter", OptimizationReporter, false);
   addTaskDependency("add_optimization_reporter", "add_reporter");
-
-  HeatConductionApp::registerAll(f, af, s);
-  StochasticToolsApp::registerAll(f, af, s);
```

Comment:
@lynnmunday , you removed teh Stochastic Tools registration here. That's why the action doesn't work.

Reply:
@dschwen  should I put it back in or make my own for the optimization module?